File: Hash Table/349. Intersection of Two Arrays/intersection.cpp
class Solution
{
public:
    vector<int> intersection(vector<int> &nums1, vector<int> &nums2)
    {
        vector<bool> hashset;
        vector<int> intersect;
        hashset.resize(1001);
        fill(hashset.begin(), hashset.end(), false);

        for (auto item : nums1)
        {
            if (hashset[item] == false)
            {
                hashset[item] = true;
            }
        }

        for (auto item : nums2)
        {
            if (hashset[item] == true)
            {
                hashset[item] = false;
                intersect.push_back(item);
            }
        }
        return intersect;
    }
};

File: 4. Hash Table/349. Intersection of Two Arrays/intersection.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        hashset = [False] * 1001
        intersect = []

        for item in nums1:
            if hashset[item] == False:
                hashset[item] = True

        for item in nums2:
            if hashset[item] == True:
                hashset[item] = False
                intersect.append(item)

        return intersect